packages/tailwindcss/src/compat/plugin-api.ts

```diff
@@ -212,7 +220,9 @@ export function buildPluginApi(
 
       for (let [name, css] of entries) {
         if (name.startsWith('@keyframes ')) {
-          ast.push(rule(name, objectToAst(css)))
+          if (!referenceMode) {
+            ast.push(rule(name, objectToAst(css)))
+          }
```

Comment:
This was the place where previously it's possible to sneak-through `@keyframes` even when the plugin is inside an `@reference` import